### PR TITLE
Using max_stake_wanted

### PR DESCRIPTION
**Pull request by ochaloup**
Contract, SDK and CLI change for the usage of `max-wanted-stake` in `Bond` configuration.

I do create this as a draft PR but it should work fine as it's now coded. However, I would like to do the first round of review that clarifies that the contract code does what it is expected to do from the perspective of the delegation strategy.
Especially it's about `Bond` fields `cpmpe` and `max-stake-wanted` and the field in ` Config` that defines minimum value for `max-stake-wanted`.

### CODE REVIEW

**Comment by ochaloup** [comment]
The current code requires fixing of the `README`. That's in progress now.

**Comment by ochaloup** [comment]
The `README` and cli output are updated. So moving to normal PR.

File: programs/validator-bonds/src/checks.rs [hunk 1]
```diff
@@ -252,7 +252,8 @@ mod tests {
             authority,
             cpmpe: 0,
             bump: 0,
-            reserved: [0; 142],
```
[the remark this answers is not in the document]
Comment (ochaloup) [reply programs/validator-bonds/src/checks.rs:255]: This is a leftover from time I thought we would have min and max values and I forgot to fix it. Thanks for finding this err :+1:

File: packages/validator-bonds-cli/src/commands/show.ts [hunk 2]
```diff
@@ -469,21 +470,75 @@ function reformatBond(key: string, value: any): ReformatAction {
   ) {
     return { type: 'Remove' }
   }
-  if (key.toLowerCase() === 'cpmpe' || key.toLowerCase().includes('bump')) {
-    return { type: 'Remove' }
+  if (key.toLowerCase().includes('cpmpe')) {
```
[the remark this answers is not in the document]
Comment (ochaloup) [reply packages/validator-bonds-cli/src/commands/show.ts:473]: I would need help with some ideas here.  Let's elaborate.

I intend to separate the way data is stored and how they are displayed.  This is something that I don't want to change and I need to find then a better way how to approach this.

I went with the most straightforward way to do so. Using string comparison makes an easy to make changes for not providing some direct typing. Or there is a better way to do so.

Using the string comparison makes it easy to work with some "set" of values to be displayed in some specific way - like when 'amount' then prints with `SOL` etc. Having it typed could bring more boilerplate but a) maybe not, and b) I'm not against that apriory.

Would you know about some TS approach for this or a library that solves this type of handling?

**Review by janlegner — APPROVED** [review]

**Pull request merged**